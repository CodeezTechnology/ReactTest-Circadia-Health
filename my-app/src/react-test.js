import React from 'react';
import axios from 'axios'
import LoadingScreen from 'react-loading-screen';

export class CalculateSquareRoot
    extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            value: '',
            number: '',
            loading: false
        };
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(event) {
        this.setState({
            value: event.target.value
        });
    }

    handleSubmit(event) {
        event.preventDefault();
        if (this.state.value !== '' && this.state.value !== null) {
            console.log("number " + this.state.value);
            this.setState({
                loading: true
            });
            axios.get(`http://localhost:5000/${this.state.value}`)
                .then(response => {
                    this.setState({ number: JSON.stringify(response.data.data) });
                    //debugger;
                    this.setState({
                        loading: false
                    });
                    console.log(response.data.data);
                })
                .catch(function (error) {
                    console.log(error);
                })
        }
        else {
            alert('Please enter any number');
        }

    }
    render() {
        return (
            /* eslint-disable */
            /* eslint-disable no-unused-expressions */
            <LoadingScreen
                loading={this.state.loading ? true : false}
                bgColor='#f1f1f1'
                spinnerColor='blue'
                textColor='#676767'
                logoSrc='/logo192.png'
                text='Please wait Square root is being calculating..'>

                <div>
                    <h3>React Test</h3>
                    <form onSubmit={this.handleSubmit} id="fmSubmit">
                        <label>  Number:
                <input
                                id="txtNumber"
                                type="number"
                                className="input-user"
                                placeholder="number"
                                value={this.state.value}
                                onChange={this.handleChange}
                                min="0" step="1"
                            />
                        </label>
                        <input
                            id="btnSubmit"
                            type="submit"
                            value="calculate square root"
                        />
                        <hr></hr>
                        <div>
                            <h1 id="txtResult">{this.state.number}</h1>
                        </div>
                    </form>
                </div>
            </LoadingScreen>
        );
    }
}

