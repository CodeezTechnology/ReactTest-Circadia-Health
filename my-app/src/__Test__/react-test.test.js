import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import { shallow, configure } from "enzyme";
import { CalculateSquareRoot } from '../react-test'
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

it("renders without crashing", () => {
    shallow(<CalculateSquareRoot />);
});

it("renders header", () => {
    const wrapper = shallow(<CalculateSquareRoot />);
    const label = <h3>React Test</h3>;
    expect(wrapper.contains(label)).toEqual(true);
});

describe('Test Button component', () => {
    it('Test click event', () => {
        const mockCallBack = jest.fn();

        const button = shallow((<CalculateSquareRoot onClick={mockCallBack}>Ok!</CalculateSquareRoot>));
        button.find('input#btnSubmit').simulate('submit');
        expect(mockCallBack.mock.calls.length).not.toEqual(1);


    });
});

it("renders correctly", () => {
    const tree = shallow(<CalculateSquareRoot />);
    expect(tree.find('input#btnSubmit').simulate('submit', { target: { value: '' } }));
});

describe('Input text box component', () => {
    it('Test Input number', () => {
        const wrapper = shallow(<CalculateSquareRoot />);
        wrapper.find('input#txtNumber').simulate('change', { target: { value: Number } });
        expect(wrapper.state('value')).toBe(Number);
        // expect(wrapper.state('value')).toBe(String); //Error occurred- Test failed

        const findInput= wrapper => wrapper.find('input#txtNumber');
        findInput(wrapper).simulate('change',{target : {value :'64'}});
        expect(findInput(wrapper).prop('value')).toBe('64');
        
    });
});

it("Input Value", () => {
    const div = document.createElement("div");
    ReactDOM.render(<CalculateSquareRoot />, div);

    const { getByText } = render(<CalculateSquareRoot />);
    const inputNumber = getByText(/number/i);
    expect(inputNumber).toBeInTheDocument();
})

describe("Square roor component", () =>{
    it("renders",()=>{
        const wrapper = shallow(<CalculateSquareRoot/>);
        expect(wrapper).toMatchSnapshot();
    });
})

it("When the form is submitted the event is cancelled", () => {
   
    const wrapper = shallow(<CalculateSquareRoot/>);
    let prevented = false;
    wrapper.find("form").simulate("submit",{
        preventDefault:()=>{
            prevented =true;
        }
    });
    expect(prevented).toBe(true);
})

    it('State Result', () => {
        const wrapper = shallow(<CalculateSquareRoot />);
        const state = {
            value: '64',
            number: '{"result":8}',
            loading: true
        }
      
       wrapper.setState(state);
       expect(wrapper.state('value')).toBeTruthy();
       expect(wrapper.state('number')).toBeTruthy();
       expect(wrapper.state('loading')).toBeTruthy();
    });

    it('Submit event fired', () => {
        const wrapper = shallow(<CalculateSquareRoot />);
        const formEventMocked = { preventDefault: jest.fn() };
        wrapper.find('form').simulate('submit', formEventMocked);
        expect(wrapper.find('form')).toHaveLength(1);
        expect(formEventMocked.preventDefault).toBeCalledTimes(1);
    });
