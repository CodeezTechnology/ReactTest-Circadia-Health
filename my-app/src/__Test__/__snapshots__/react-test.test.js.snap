// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square roor component renders 1`] = `ShallowWrapper {}`;
